channeldb/migration26/migration.go

```diff
@@ -0,0 +1,275 @@
+package migration26
+
+import (
+	"bytes"
+	"fmt"
+
+	mig25 "github.com/lightningnetwork/lnd/channeldb/migration25"
+	mig "github.com/lightningnetwork/lnd/channeldb/migration_01_to_11"
+	"github.com/lightningnetwork/lnd/kvdb"
+)
+
+// recordsPerTx specifies the number of records to be migrated in each database
+// transaction. In the worst case, each old revocation log is roughly 0.75
+// MB(747573 bytes). 1000 records would consume 0.75 GB of ram, which is
+// feasible for a modern machine.
+const recordsPerTx = 1000
```

Comment:
This figure might be way toooo conservative. I'm thinking instead of tracking the number of records, we can use `runtime.ReadMemStats` to monitor and cap the memory usage.

Reply:
Decided to convert the old large commitments into revocation logs first to save more memory. However, as the benchmark has shown, the CPU was not bounded by this memory usage at all. Rather, most of the CPU time was taken by `DeriveCommitmentKeys` that's used to calculate the output indexes.